packages/server/src/modules/step/middleware.ts

```diff
@@ -0,0 +1,81 @@
+import { MiddlewareFunction } from '@types';
+import { Step } from '@chess-tent/models';
+import { StepNotFoundError, UnauthorizedStepEditError } from './errors';
+import * as service from './service';
+
+export const saveStep: MiddlewareFunction = (req, res, next) => {
+  service
+    .saveStep(res.locals.step as Step)
+    .then(next)
+    .catch(next);
+};
+
+export const publishStep: MiddlewareFunction = (req, res, next) => {
```

Comment:
Following the comment above, let's drop logic related publishing.

Reply:
👍